clang/lib/3C/RewriteUtils.cpp

```diff
@@ -491,14 +505,20 @@ class TypeArgumentAdder : public clang::RecursiveASTVisitor<TypeArgumentAdder> {
 };
 
 SourceRange FunctionDeclReplacement::getSourceRange(SourceManager &SM) const {
-  SourceLocation Begin = RewriteReturn ? getDeclBegin(SM) : getParamBegin(SM);
+  SourceLocation Begin = RewriteGeneric ? getGenericBegin(SM) :
+                      (RewriteReturn ? getDeclBegin(SM) : getParamBegin(SM));
   SourceLocation End = RewriteParams ? getDeclEnd(SM) : getReturnEnd(SM);
   // Begin can be equal to End if the SourceRange only contains one token.
   assert("Invalid FunctionDeclReplacement SourceRange!" &&
          (Begin == End || SM.isBeforeInTranslationUnit(Begin, End)));
   return SourceRange(Begin, End);
 }
 
+SourceLocation FunctionDeclReplacement::getGenericBegin(SourceManager &SM) const {
+  SourceLocation Begin = Decl->getSourceRange().getBegin();
```

Comment:
I think you're right and this was changed before I realized we needed RewriteGeneric to imply RewriteReturn. I'll see if I can remove this.

Reply:
Nope, this is actually an existing bug that I might as well fix here.
```
_For_any(T) int *viewer(T *i, int *j) { return j; }
```
converts to
```
_Ptr<int> viewer(_Ptr<T> i, _Ptr<int> j) _Checked { return j; }
```
Losing its forany, because the rewrite starts too early. I'll change it.

Reply:
I gave up and left comments. It's an existing issue that requires partially-converted code.